platform/pulpcore/app/viewsets/repository.py

```diff
@@ -141,7 +141,7 @@ def destroy(self, request, repository_name, name):
         )
         return OperationPostponedResponse([async_result], request)
 
-    @decorators.detail_route()
+    @decorators.detail_route(methods=('post',))
```

Comment:
Internally to drf (and in their examples) this is a list rather than a tuple. I'm sure this works, but I would use a list, safety first :) 

I won't block on this though. Change if you want.